resqpy/well/_blocked_well.py

```diff
@@ -3154,3 +3154,58 @@ def write_hdf5(self, file_name = None, mode = 'a', create_for_trajectory_if_need
         h5_reg.register_dataset(self.uuid, 'LocalFacePairPerCellIndices', raw_face_indices)  # could use uint8?
 
         h5_reg.write(file = file_name, mode = mode)
+
+    def add_grid_property_to_blocked_well(self, uuid_list):
+        """Add properties to blocked wells from a list of uuids for properties on the supporting grid."""
+
+        part_list = [self.model.part_for_uuid(uuid) for uuid in uuid_list]
+
+        assert len(self.grid_list) == 1, "Only blocked wells with a single grid can be handled currently"
+        grid = self.grid_list[0]
+        parts = self.model.parts_list_filtered_by_supporting_uuid(part_list,
+                                                                  grid.uuid)  # only those properties on the grid
+        gridpc = grid.property_collection
+        parts = [part for part in parts if gridpc.indexable_for_part(part) == 'cells'
+                ]  # only 'cell' properties are handled
+
+        if len(parts) > 0:
+            bwpc = rqp.PropertyCollection(support = self)
+            if len(gridpc.time_series_uuid_list()) > 0:
+                time_dict = {}
+                for part in parts:
+                    if gridpc.time_series_uuid_for_part(part) in time_dict.keys():
+                        time_dict[gridpc.time_series_uuid_for_part(
+                            part)] = time_dict[gridpc.time_series_uuid_for_part(part)] + [part]
+                    else:
+                        time_dict[gridpc.time_series_uuid_for_part(part)] = [part]
+            else:
+                time_dict = {None: parts}
```

Comment:
Hmm, unless I'm missing something, this coding will skip static properties if there are one or more time varying properties present, which is not ideal. I would suggest building a temp list of parts that you have assigned to any entry in the time_dict as you build it up, then in this final else you can simply set that temp list to the empty list [ ]. After that, outside the if else, you could set the time_dict entry for None to be the list of parts that are not in that temp list.

Reply:
I stand corrected.